aten/src/ATen/native/BucketizationUtils.h

```diff
@@ -134,6 +134,13 @@ inline void searchsorted_pre_check(
 
     TORCH_CHECK(sorter.scalar_type() == ScalarType::Long, "torch.searchsorted(): sorter must be a tensor of long ",
       "dtype but got dtype ", sorter.scalar_type());
+
+    if (sorter.numel() > 0) {
+      auto minmax = sorter.aminmax();
+      long vmin = std::get<0>(minmax).item().toLong();
```

Comment:
is `long` here int64_t? or equal to int? should it be explicitly said int32_t or int64_t?

Reply:
`toLong()` returns exactly `long` whose bit width is platform-defined to be at least 32bit (https://en.cppreference.com/w/cpp/language/types). On Linux x64 it is 8 bytes (https://gcc.godbolt.org/z/bb96x1d1K).

Reply:
Oh, after further investigation it seems to be `int64_t` (displayed to be long as it is equivalent to long on my platform). https://github.com/pytorch/pytorch/blob/3779a75fc9ff92f3f3bc6409264764a5cd801201/c10/core/ScalarType.h#L70

Yeah, we should explicitly make it `int64_t` as you suggested. Thanks!